processor/src/main/java/org/mapstruct/ap/internal/model/source/BeanMappingOptions.java

```diff
@@ -55,15 +55,26 @@ public static BeanMappingOptions forInheritance(BeanMappingOptions beanMapping,
         return options;
     }
 
-    public static BeanMappingOptions forForgedMethods(BeanMappingOptions beanMapping) {
-        BeanMappingOptions options = new BeanMappingOptions(
+    public static BeanMappingOptions forForgedMethods(MappingMethodOptions mappingMethodOptions) {
+
+        BeanMappingOptions beanMapping = mappingMethodOptions.getBeanMapping();
+
+        return new BeanMappingOptions(
             beanMapping.selectionParameters != null ?
                 SelectionParameters.withoutResultType( beanMapping.selectionParameters ) : null,
-            Collections.emptyList(),
+            takeOverIgnoreUnmappedSourceProperties( mappingMethodOptions ),
```

Comment:
The easy fix would've been to just take over the `beanMapping.getIgnoreUnmappedSourceProperties()`. However this led to an error in the `Issue2743Test` (`Issue2743Mapper`)...